### PR TITLE
arm64singlestepper

### PR DESCRIPTION
**Pull request by sdmaclea**
In progress work to enable arm64 single step emulation

/cc @noahfalk @tommcdon @hoyosjs

### CODE REVIEW

**Comment by sdmaclea** [comment]
I think this PR is now functionally complete.

I am going to start debugging.  It would be nice to review it concurrently with debugging given the need to get this in preview8.

File: src/debug/ee/controller.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ bool DebuggerControllerPatch::IsSafeForStackTrace()
 
 }
 
-#ifndef _TARGET_ARM_
+#if !(defined(_TARGET_ARM_) || (defined(_TARGET_ARM64_) && defined(PLATFORM_UNIX)))
```
Comment (jkotas) [inline src/debug/ee/controller.cpp]: This condition is repeated many times. Would it make sense to create a new defined for this?

Comment (sdmaclea) [reply src/debug/ee/controller.cpp]: I'll add `FEATURE_EMULATE_SINGLESTEP`

Comment (hoyosjs) [reply src/debug/ee/controller.cpp]: Isn't that counter intuitive? This is is the define that asks if single step is hardware supported.

Comment (sdmaclea) [reply src/debug/ee/controller.cpp]: Counter intuitive? We are adding support for `arm64 single step emulation` why would `FEATURE_EMULATE_SINGLESTEP` be counter-intuitive as a feature name.

It obviously has to be used correctly as needed
```C++
#ifdef FEATURE_EMULATE_SINGLESTEP
```
```C++
#ifndef FEATURE_EMULATE_SINGLESTEP
```

Comment (sdmaclea) [reply src/debug/ee/controller.cpp]: Done

Comment (hoyosjs) [reply src/debug/ee/controller.cpp]: Oh ok, so the positive of the condition. Thanks steve :)

File: src/debug/ee/arm64/primitives.cpp [hunk 2]
```diff
@@ -52,4 +52,5 @@ bool IsSSFlagEnabled(DT_CONTEXT *pContext, Thread *)
 {
     return IsSSFlagEnabled(pContext);
 }
+#error zdfglhjksd
```
Comment (hoyosjs) [inline src/debug/ee/arm64/primitives.cpp]: Is this intended?

Comment (sdmaclea) [reply src/debug/ee/arm64/primitives.cpp]: It was stray debugging code that accidentally got checked in._

**Comment by sdmaclea** [comment]
The basics of this are working.  I will continue to poke on it today then merge in a few hours.

**Comment by tommcdon** [comment]
This PR should fix #25107

**Pull request merged**